axelrod/strategies/armed_bandits.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+
+from axelrod.action import Action
+from axelrod.player import Player
+
+C, D = Action.C, Action.D
+
+
+class Greedy(Player):
+    """
+    A player that always chooses the optimal action based on the average reward of each action from previous turns.
+
+    If initial rewards for each action are equivalent (true by default),
+    then the optimal action for the first turn is cooperate.
+
+    Names:
+
+    - Greedy: [Sutton2018]_
+    """
+
+    name = "greedy"
+    classifier = {
+        "memory_depth": float("inf"),
+        "stochastic": False,
+        "long_run_time": False,
+        "inspects_source": False,
+        "manipulates_source": False,
+        "manipulates_state": False,
+    }
+
+    UNIFORM = np.inf  # constant that replaces weight when rewards aren't weighted
```

Comment:
I've changed this to -1.0, and changed other places in the code to refer to this constant for consistency. This does mean that if a user uses recency_wieght=-1.0 at time of creation, it will be treated as not recency weighted (instead of an out of range value limited to 0.0 as in previous implementations).